File: Proyecto_1/src/Error/ListError.java
package Error;

import java.util.ArrayList;

public class ListError {
    private final ArrayList<Error> listError = new ArrayList<>();

    private static final ListError instance = new ListError();

    public static ListError getInstance() {
        return instance;
    }

    public ArrayList<Error> getListError() {
        return listError;
    }

    public void addError(Error e) {
        listError.add(e);
    }

    public boolean isEmpty() {
        return listError.isEmpty();
    }

    public void clearList() {
        listError.clear();
    }
}

File: Proyecto_2/Back/API/Parser/Error/ListError.js
class ListError {
    listaErrores = [];

    static instance = new ListError();

    static getInstance() {
        return this.instance;
    }

    addLista(error) {
        this.listaErrores.push(error);
    }

    getLista() {
        return this.listaErrores;
    }

    clearLista() {
        this.listaErrores = [];
    }
}

module.exports = ListError;